### PR TITLE
nf-validation plugin for samplesheet validation

### PR DESCRIPTION
**Pull request by LilyAnderssonLee**
Close issues #3 #5 
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/metaval/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/metaval _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by LilyAnderssonLee** [comment]
The raw test data was downloaded from the article [A streamlined clinical metagenomic sequencing protocol for rapid pathogen identification](https://www.nature.com/articles/s41598-021-83812-x) including three Illumina samples and one Nanopore sample.
Illumina samples: SRR13439790; SRR13439802; SRR13439813
Nanopore sample: SRR13439799

File: assets/samplesheet.csv
Comment (sofstam) [inline assets/samplesheet.csv:1]: 🚀 
1. Why is `reads_type` needed as an extra column? Wouldn't it be inferred by `instrument_platform`?
2. Are both output files from kraken2 and centrifuge needed to have information from taxid, right?

File: workflows/metaval.nf [hunk 1]
```diff
@@ -21,10 +21,10 @@ WorkflowMetaval.initialise(params, log)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-ch_multiqc_config          = Channel.fromPath("$projectDir/assets/multiqc_config.yml", checkIfExists: true)
-ch_multiqc_custom_config   = params.multiqc_config ? Channel.fromPath( params.multiqc_config, checkIfExists: true ) : Channel.empty()
-ch_multiqc_logo            = params.multiqc_logo   ? Channel.fromPath( params.multiqc_logo, checkIfExists: true ) : Channel.empty()
-ch_multiqc_custom_methods_description = params.multiqc_methods_description ? file(params.multiqc_methods_description, checkIfExists: true) : file("$projectDir/assets/methods_description_template.yml", checkIfExists: true)
+//ch_multiqc_config          = Channel.fromPath("$projectDir/assets/multiqc_config.yml", checkIfExists: true)
```
Comment (sofstam) [inline workflows/metaval.nf]: Shall those just be removed or what do you think?

File: workflows/metaval.nf [hunk 2]
```diff
@@ -35,7 +35,7 @@ ch_multiqc_custom_methods_description = params.multiqc_methods_description ? fil
 //
 // SUBWORKFLOW: Consisting of a mix of local and nf-core/modules
 //
-include { INPUT_CHECK } from '../subworkflows/local/input_check'
+//include { INPUT_CHECK } from '../subworkflows/local/input_check'
```
Comment (sofstam) [inline workflows/metaval.nf]: Same here

File: workflows/metaval.nf [hunk 3]
```diff
@@ -46,9 +46,9 @@ include { INPUT_CHECK } from '../subworkflows/local/input_check'
 //
 // MODULE: Installed directly from nf-core/modules
 //
-include { FASTQC                      } from '../modules/nf-core/fastqc/main'
-include { MULTIQC                     } from '../modules/nf-core/multiqc/main'
-include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoftwareversions/main'
+//include { FASTQC                      } from '../modules/nf-core/fastqc/main'
```
Comment (sofstam) [inline workflows/metaval.nf]: Here as well.

File: workflows/metaval.nf [hunk 4]
```diff
@@ -66,21 +66,23 @@ workflow METAVAL {
...
     // ! There is currently no tooling to help you write a sample sheet schema
 
     //
     // MODULE: Run FastQC
     //
-    FASTQC (
-        INPUT_CHECK.out.reads
-    )
-    ch_versions = ch_versions.mix(FASTQC.out.versions.first())
+    //FASTQC (
```
Comment (sofstam) [inline workflows/metaval.nf]: Can this be removed or is this default by the template?

**Review by sofstam — CHANGES_REQUESTED** [review]
Nice! Good to set up some actions? Similar to taxprofiler or https://github.com/genomic-medicine-sweden/tomte/actions.

File: assets/samplesheet.csv
Comment (LilyAnderssonLee) [reply assets/samplesheet.csv:1]: Thanks for those suggestions. I will remove `reads_type` and will utilise the information of `instrument_platform`. 
Yes, we need both output files from kraken2 and centrifuge.

File: workflows/metaval.nf (hunk 1 above)
Comment (LilyAnderssonLee) [reply workflows/metaval.nf]: Removed.

File: workflows/metaval.nf (hunk 4 above)
Comment (LilyAnderssonLee) [reply workflows/metaval.nf]: It's the default by the template. I have removed it since I will not use it in the pipeline.

**Comment by LilyAnderssonLee** [comment]
I have enabled github Action and will add CI test later.

**Pull request closed without merging**